puppeteer-test/src/tests/navigation/navigationmenu.spec.ts

```diff
@@ -0,0 +1,68 @@
+import GoogleLoginPage from '../../pages/google-login';
+import Home from '../../pages/home';
+import Profile from '../../pages/profile';
+import Workspaces from '../../pages/workspaces';
+import PuppeteerLaunch from '../../services/puppeteer-launch';
+
+jest.setTimeout(60 * 1000);
+
+const configs = require('../../config/config');
+
+describe('Navigation menu tests:', () => {
+
+  let browser;
+  let incognitoContext;
+  let page;
+
+  beforeAll(async () =>  {
+    browser = await PuppeteerLaunch();
+  });
+
+  beforeEach(async () => {
+    incognitoContext = await browser.createIncognitoBrowserContext();
```

Comment:
Is the Incognito context piece important here? It's not used for the homepage spec, so I'm wondering about the inconsistency. If I'm a RW dev planning to write a new spec for my new feature, which pattern should I prefer? Documenting the rationale behind this choice somewhere would be helpful – this could equally end up in a "base" test case wrapper file, in the README, or as a comment in one of these files.

Reply:
I recommend all tests run in Incognito mode to prevent tests sharing any cookies or cache with the other tests, especially true if running in parallel. The Homepage spec is a very simple version of example test created at the beginning. I will remove Homepage spec very soon in near future.